### ISSUE TITLE
Very low unit test coverage

### ISSUE THREAD

**Issue by akhil-ghatiki** [nic-delhi/AarogyaSetu_Android#82]
The over all code have very low test coverage. 
The PRs merged can potentially break the functionality. Its very important to bring up the unit test cases.

**Comment by yashwant121** [comment]
Yes, I totally agree with you as more unit test cases should be there. So I would like to take up this issue.

**Comment by siddhpant** [comment]
+1. More test coverage is always desirable. Though I think we should cut some slack for the devs, they probably whipped up the app on a short notice/period of time considering the situation.

**Comment by akhil-ghatiki** [comment]
@yashwant121 we can split this issue into smaller pieces. Also please raise small PRs so that we can get quick feedback

**Comment by akhil-ghatiki** [comment]
@siddhpant +1. slack will help in discussing.

**Comment by yashwant121** [comment]
AarogyaSetu_Android/tree/master/app/src/androidTest/java/nic/goi/aarogyasetu/utility

I can see three files here, So lets take one-one file each . It will be easy that way.

**Comment by akhil-ghatiki** [comment]
@yashwant121 are they even merging the PRs ? I see a lot many open PRs without any discussion. 
The PRs that are merged after this code is open soured are only a hand few.

**Comment by siddhpant** [comment]
> yashwant121 are they even merging the PRs ? I see a lot many open PRs without any discussion.
> The PRs that are merged after this code is open soured are only a hand few.

@akhil-ghatiki They won't. Open-sourcing code doesn't imply they are requesting contributions. It is open-sourced in public interest. Also, no PR is merged till date.

**Comment by akhil-ghatiki** [comment]
@siddhpant The whole point of open sourcing is lost in that case. Who knows if they are actually deploying this code or not ?

**Comment by siddhpant** [comment]
> @siddhpant The whole point of open sourcing is lost in that case. Who knows if they are actually deploying this code or not ?

@akhil-ghatiki That's the case with every open-source software of which you don't have build/publisher access, regardless of you being a contributor to that project.

Besides, the point of open-sourcing is not that of allowing public contribs, rather it is to allow access to source code so that anyone can study and analyse it, and in many cases modify and/or redistribute it (though subject to license), IMO. 

You have access to the source. You can analyse the code, and point out the flaws, as pointed out in README. There's also a bug-bounty programme. 

Besides, you can decompile the APK and check.

**Comment by akhil-ghatiki** [comment]
@siddhpant I know this. I am not asking for access to build/publisher. I am pretty much aware of open source world, made few contributions earlier as well. The PRs merged will actually go into the build and the deployed software.(of course, contributors won't have control over it).

I would differ on your opinion on the point of open sourcing. Open sourcing the code is much more than just allowing the public to see the code. It is about taking the contributions from community and making it a better, reliable, usable,transparent code/software. Take example of any great open sourced software today the world is using. Spark, kafka, JS libs... and a million more. All these reached to that point because of the community contributions because open source contributors may have other cases they would solve which the creators of the software never thought off. 

One of the true signs of a healthy open sourced software is not just people using and seeing it but people actually contributing to that code. It is more like a social movement. 

The power of open source is the power of people. The people own it and rule it.

**Comment by siddhpant** [comment]
@akhil-ghatiki I fully agree with the general idea of what you said, but what I was saying was technically/pedantically "open-source" software need not necessarily require to allow public contribution to it. Besides, the maintainers of this project accept critical suggestions and flaws affecting it on their email. I would like if they start accepting PRs they find worthy, though.

Also, I didn't say you are asking for build access. It was a reply to "how do we know if same thing that is here is published or not".
